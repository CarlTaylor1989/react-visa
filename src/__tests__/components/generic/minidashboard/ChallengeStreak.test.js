/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import {
  ChallengeStreak,
  mapStateToProps
} from '../../../../components/generic/minidashboard/ChallengeStreak';

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    progress: 0
  };
});

test('renders the ChallengeStreak component', () => {
  const wrapper = shallow(<ChallengeStreak {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders the ChallengeStreak component with progress of two', () => {
  const wrapper = shallow(<ChallengeStreak {...mockProps} progress={2} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('should mapStateToProps', () => {
  const state = {
    streakData: {
      progress: 0
    }
  };

  const props = mapStateToProps(state);
  expect(props).toEqual({
    ...state.streakData
  });
});
