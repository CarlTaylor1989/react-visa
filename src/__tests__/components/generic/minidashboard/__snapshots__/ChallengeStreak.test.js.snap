// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ChallengeStreak component 1`] = `
<div
  className="challengeStreak"
>
  <Translate
    tag="span"
    value="generic.miniDashboard.bonusStreak"
  />
  <div
    className="streakContainer"
  >
    <div
      className="streakCircle "
      key="1"
    />
    <div
      className="streakCircle "
      key="2"
    />
    <div
      className="streakCircle "
      key="3"
    />
    <div
      className="streakCircle "
      key="4"
    />
    <Translate
      className="bonus"
      tag="span"
      value="generic.miniDashboard.bonus"
    />
  </div>
</div>
`;

exports[`renders the ChallengeStreak component with progress of two 1`] = `
<div
  className="challengeStreak"
>
  <Translate
    tag="span"
    value="generic.miniDashboard.bonusStreak"
  />
  <div
    className="streakContainer"
  >
    <div
      className="streakCircle filled"
      key="1"
    />
    <div
      className="streakCircle filled"
      key="2"
    />
    <div
      className="streakCircle "
      key="3"
    />
    <div
      className="streakCircle "
      key="4"
    />
    <Translate
      className="bonus"
      tag="span"
      value="generic.miniDashboard.bonus"
    />
  </div>
</div>
`;
